# 求助！哪位老师帮助看一下，这个提示如何处理？

执行程序总有查询超时错误。可是我单独执行那条语句就没问题，可以执行。请教老师这是怎么回事？如何处理？

date_bg = u"20070101"
date_ed = u"20180101"

Factor = {u"ROE":[u"权益回报率",u"权益回报率=净利润（TTM）/股东权益"],
         u"NPParentCompanyGrowRate":[u"归属母公司股东的净利润增长率",u"归属母公司股东的净利润增长率=(今年归属于母公司所有者的净利润（TTM）/去年归属于母公司所有者的净利润（TTM）)-1"],
         u"CashRateOfSales":[u"经营活动产生的现金流量净额与营业收入之比",u"经营活动产生的现金流量净额与营业收入之比=经营活动产生的现金流量净额（TTM）/营业收入"]}

ticks_list = list(ticks_df[u'ticker'])
ticker_field = [u"tradeDate"]+Factor.keys()
ticker_ans = pd.DataFrame()

progress = []
print "进度：",
for t in ticks_list:

    ticker_fac = DataAPI.MktStockFactorsDateRangeGet(ticker=t,beginDate=date_bg,endDate=date_ed,field=ticker_field,pandas="1")
    ticker_fac = ticker_fac.dropna()           # 剔除 Nan 值
    ticker_name = ticks_df.secShortName[ticks_df.ticker == t].values[0]
    
    
    ——————————————————————————————————————————————————————————————————————————————————————————————
    Exception                                 Traceback (most recent call last)
&lt;mercury-input-7-444D28568CDF40F6BFFF8A12EEEF8157&gt; in &lt;module&gt;()
     16 for t in ticks_list:
     17 
---&gt; 18     ticker_fac = DataAPI.MktStockFactorsDateRangeGet(ticker=t,beginDate=date_bg,endDate=date_ed,field=ticker_field,pandas="1")
     19     ticker_fac = ticker_fac.dropna()           # 剔除 Nan 值
     20     ticker_name = ticks_df.secShortName[ticks_df.ticker == t].values[0]

/home/ipython/anaconda/lib/python2.7/site-packages/DataAPI/DATAYES.pyc in MktStockFactorsDateRangeGet(secID, ticker, beginDate, endDate, field, pandas)
   2230         requestString.append(field)
   2231     if split_param is None:
-&gt; 2232         csvString = api_base.__getCSV__(''.join(requestString), httpClient, gw=False)
   2233         if csvString is None or len(csvString) == 0 or (csvString[0] == '-' and not api_base.is_no_data_warn(csvString, False)) or csvString[0] == '{':
   2234             api_base.handle_error(csvString, 'MktStockFactorsDateRangeGet')

/home/ipython/anaconda/lib/python2.7/site-packages/DataAPI/api_base.pyc in __getCSV__(requestString, httpClient, gw)
    103     except ReadTimeout:
    104         logging.error('timeout for %s' % requestString)
--&gt; 105         raise Exception(get_real_string('查询服务超时'))
    106     except Exception as e:
    107         import traceback

Exception: 查询服务超时
——————————————————————————————————————————————————————————————————————————————
单独执行下面这句话，就没问题。
ticker_fac = DataAPI.MktStockFactorsDateRangeGet(ticker=t,beginDate=date_bg,endDate=date_ed,field=ticker_field,pandas="1")
ticker_fac
